### DISCUSSION TITLE
How is the redirect URL to the lender shared with the BAP?

### DISCUSSION THREAD

**Discussion by ankitagarwaladobe** [ONDC-Official/ONDC-FIS-Specifications#13] [Q&A]
It seems that eNach/SI will be setup on the lender platform. Where in the API the redirect URL shared by the lender?

What are the other cases where the user would need to redirected to the lender?

**Comment by Antriksh-Parmar** [comment]
KYC, Repayment setup, Potentially for loan agreement as well.

**Reply by ankitagarwaladobe** [reply]
@92shreyansh 
 Where in the API the redirect URL shared by the lender in above cases?

**Reply by ankitagarwaladobe** [reply]
@92shreyansh 
Also, how does the lender communicate the status of the user actions after the user is redirected on the lender portal?

For example, when the user is redirected to lender portal for KYC then when and how does the lender communicate back to the BAP about state of KYC. Submitted, under review, successful?

Will there be any callback from the lender or the BAP has to keep polling?

**Reply by 92shreyansh** [reply]
1. The redirect urls will be part of the same xinput where the form url are present
2. For status update from BPP to BAP for a form/redirect can be communicated using on_status API
3. To poll this we will use the status API, Will need to identify the refrencing identifier to poll a status as in all previous use case these APIs were used in Post Confirm where order_id was used.

**Comment by ankitagarwaladobe** [comment]
_For status update from BPP to BAP for a form/redirect can be communicated using on_status API_

IIUC we should not use _status_ and _on_status_ APIs to check the status of non transactional apis. Is it a recommendation in beckn apis as well?
